src/app/stilling/_data/adData.ts

```diff
@@ -2,29 +2,42 @@ import { containsEmail, extractEmail, isValidEmail, mailtoInString } from "@/app
 import DOMPurify from "isomorphic-dompurify";
 import fixLocationName from "@/app/_common/utils/fixLocationName";
 import logger from "@/app/_common/utils/logger";
+import {
+    AdDTORAW,
+    ContactDTO,
+    ElasticSearchAdResult,
+    EmployerDTO,
+    Location,
+    MapedAdDTO,
+    UrlDTO,
+} from "@/app/stilling/_data/types";
 
 /**
  *  --------------------------- Common Functions ---------------------------
  */
-function getString(value) {
+function getString(value: unknown): string | undefined {
     if (value && typeof value === "string") {
         return value;
     }
     return undefined;
 }
 
```

Comment:
Mange av disse funksjonene er jo bare for å sjekke om verdien er string, er det ikke bedre måte å gjøre dette på i Typescript?

Reply:
Ja nå har jeg bare konvertert alt, i utgangspunktet så er det vel egentlig ikke noe poeng å ha den funksjonen der, i de mappingene, fordi dataene vi får er jo en string eller undefined om det ikke er satt i responsen, så egentlig kan man droppe den, og heller kun ha mapping på de spesielle casene

Reply:
@khtdiep Var det ikke sånn at disse funksjonene er en slags validering av data vi mottar? At dataene har forventet type, og hvis de ikke er av forventet type, så blir de ikke inkludert dataobjektet vi returnerer? Fordi noen av dataene vi får fra backend er litt "dynamiske" og kan potensielt endre type, slik at der vi f.eks forventet er en string, så kan dukke opp object eller array? Og at vi da kan få trøbbel ellers i frontend-koden

Reply:
Har nå fjernet alle de mapping funksjonene, default verdier fra elastic search er nullable, slik at det er string undefined. trenger ikke fuksjon som sjekker på om det er string eller ikke for å å returnere undefined da det er default om det ikke finnes. har også fjernet andre funksjoner som ikke gjør annet enn å returnere data man får fra respons eller undefined.

Reply:
Er ganske sikker på at dette ikke er en sjekk bare for å teste om string er null eller ikke, men også for å sjekke at string faktisk er en string, og ikke en array eller objekt. Eller at andre data kommer men andre type enn vi forventer. Så tar vi vekk disse funksjonene, så tar vi også vekk denne sjekken.